### PR TITLE
Testing client CI

### PR DESCRIPTION
**Pull request by theolebever**
Description:

This PR adds a CI  pipeline to test the client side of BastionLab. The pipeline includes the following steps:

- Build and installation of the client
- Wait for the build of the server, download the artifact and launch the server through Docker
- Install the necessary dependencies for the testing of the client
- Testing with flake8 
- Testing with mypy
- Testing with pytest-cov
- Upload of the report made by pytest-cov

The benefits will be to no longer regress in the developpement of BastionLab client and automate the testing.

### CODE REVIEW

**Comment by JoFrost** [comment]
Is the testing with Mypy necessary? Can't flake8 do what Mypy is doing?

**Comment by theolebever** [comment]
> Is the testing with Mypy necessary? Can't flake8 do what Mypy is doing?

As I have understood, flake8 does more of a static analysis and mypy does more of a hybrid analysis. This CI was based on what has been done on BastionAI previously. But i can totally remove one of the tests if it is not necessary.

**Comment by JoFrost** [comment]
> > Is the testing with Mypy necessary? Can't flake8 do what Mypy is doing?
> 
> As I have understood, flake8 does more of a static analysis and mypy does more of a hybrid analysis. This CI was based on what has been done on BastionAI previously. But i can totally remove one of the tests if it is not necessary.

Leave it as it is, it has its usefulness.
I was mostly curious about the approach of this test.

File: bastionai/client/tests/test_client.py [hunk 1]
```diff
@@ -1,74 +1,74 @@
-from typing import Iterator
+# from typing import Iterator
```
Comment (dhalf) [inline bastionai/client/tests/test_client.py:1]: Why were these files commented out?

File: bastionai/examples/distilbert_example_notebook.ipynb [hunk 2]
```diff
@@ -45,7 +45,7 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "!pip install bastionai"
+    "!pip install bastionlab"
```
Comment (dhalf) [inline bastionai/examples/distilbert_example_notebook.ipynb:48]: Don't run examples from bastionai, there are not maintained.
Checking whether the notebooks in docs work is enough for now

File: docs/docs/how-to-guides/covid_cleaning_exploration.ipynb [hunk 3]
```diff
@@ -102,7 +102,11 @@
   {
    "cell_type": "code",
    "execution_count": 164,
-   "metadata": {},
+   "metadata": {
```
Comment (dhalf) [inline docs/docs/how-to-guides/covid_cleaning_exploration.ipynb:105]: Please double check that you don't introduce code changes in the documentation notebooks

File: examples/titanic.ipynb [hunk 4]
```diff
@@ -22,7 +22,7 @@
   },
   {
    "cell_type": "code",
-   "execution_count": 5,
+   "execution_count": 59,
```
Comment (dhalf) [inline examples/titanic.ipynb:25]: I guess this notebook is not maintained and can safely be removed

File: tests/server.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
+# import errno
```
Comment (dhalf) [inline tests/server.py:1]: This file is commented, why?

**Review by dhalf — CHANGES_REQUESTED** [review]

**Comment by dhalf** [comment]
Please resolve the conflicts with the documentation notebooks
The code MUST be the same as in master

**Review by dhalf — APPROVED** [review]

**Pull request merged**